### PR TITLE
Ports: Add Port for the yasm assembler

### PR DESCRIPTION
**Pull request by 06needhamt**
This is a required dependency for many libraries such as libx264
which i'm interested in porting in the future

### CODE REVIEW

File: Ports/yasm/package.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=yasm
+version=1.3.0
+files="https://www.tortall.net/projects/yasm/releases/yasm-${version}.tar.gz yasm-${version}.tar.gz"
```
Comment (bcoles) [inline Ports/yasm/package.sh:4]: This should use some form of integrity verification. Signature is best, otherwise a hard coded-known-good shasum.

**Comment by bcoles** [comment]
Tests failed as `yasm` wasn't added to [Ports/AvailablePorts.md](https://github.com/SerenityOS/serenity/blob/master/Ports/AvailablePorts.md):

```
 [OK]: Meta/lint-shell-scripts.sh
Running Meta/lint-ports.py... 
AvailablePorts.md is missing the following ports:
    yasm
[FAIL]: Meta/lint-ports.py
```

File: Ports/yasm/package.sh (hunk 1 above)
Comment (06needhamt) [reply Ports/yasm/package.sh:4]: I couldn't find a signature for it so I added it manually. I hope I did it right

> Tests failed as `yasm` wasn't added to [Ports/AvailablePorts.md](https://github.com/SerenityOS/serenity/blob/master/Ports/AvailablePorts.md):
> 
> ```
>  [OK]: Meta/lint-shell-scripts.sh
> Running Meta/lint-ports.py... 
> AvailablePorts.md is missing the following ports:
>     yasm
> [FAIL]: Meta/lint-ports.py
> ```

I have now added this

Comment (06needhamt) [reply Ports/yasm/package.sh:4]: I couldn't find a signature so I added it manually I hope I did it right.

**Comment by 06needhamt** [comment]
> Tests failed as `yasm` wasn't added to [Ports/AvailablePorts.md](https://github.com/SerenityOS/serenity/blob/master/Ports/AvailablePorts.md):
> 
> ```
>  [OK]: Meta/lint-shell-scripts.sh
> Running Meta/lint-ports.py... 
> AvailablePorts.md is missing the following ports:
>     yasm
> [FAIL]: Meta/lint-ports.py
> ```

I have now added this

File: Ports/yasm/patches/patch-configure.patch [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+-    as_fn_error $? "$SHELL $ac_aux_dir/config.sub $host_alias failed" "$LINENO" 5
+-fi
+-
+-fi
+-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_host" >&5
+-$as_echo "$ac_cv_host" >&6; }
+-case $ac_cv_host in
+-*-*-*) ;;
+-*) as_fn_error $? "invalid value of canonical host" "$LINENO" 5;;
+-esac
```
Comment (awesomekling) [inline Ports/yasm/patches/patch-configure.patch:27]: Can we solve this by patching `config.sub` instead? That's what we do for all other ports that use autotools :)

Comment (06needhamt) [reply Ports/yasm/patches/patch-configure.patch:27]: I was going to do that but there doesn't seem to be a config.sub in the tar.

Comment (awesomekling) [reply Ports/yasm/patches/patch-configure.patch:27]: Huh, all right

**Pull request merged**